crates/rolldown/tests/esbuild/ts/ts_declare_const/artifacts.snap

```diff
@@ -1,6 +1,23 @@
 ---
 source: crates/rolldown_testing/src/integration_test.rs
 ---
+# warnings
+
+## PARSE_ERROR
+
+```text
+[PARSE_ERROR] Warning: Identifier `foo` has already been declared
+   ╭─[entry.ts:5:15]
+   │
+ 5 │ declare const foo: any
+   │               ────┬───  
+   │                   ╰───── `foo` has already been declared here
+ 6 │ let foo = bar()
```

Comment:
What's the intention of these tests? It feels like we should patch these incorrect tests?

Reply:
https://github.com/rolldown/rolldown/blob/e47e8611a818d28c300160d811907c2306bcb5a4/crates/rolldown/src/utils/parse_to_ecma_ast.rs#L92-L100

https://github.com/rolldown/rolldown/blob/e47e8611a818d28c300160d811907c2306bcb5a4/crates/rolldown/src/utils/parse_to_ecma_ast.rs#L20-L28

Base on the code, they should be parsed as module.

Reply:
All snapshots that esbuild could generate output we treat it as correct, esbuild test are treated as incorrect(the input has syntax error and so on), and will not emit any snapshots.
We just align with that behavior.
bundler has better tolerance than a normal parser, we should try to recover (ignore) trivial errors and bundle something to user.

Reply:
> What's the intention of these tests? It feels like we should patch these incorrect tests?

There is no very clear incorrect definition in the bundler, so I prefer not to modified the tests.